README.adoc

```diff
@@ -1,180 +1,192 @@
+== Improving startup time
 
-The common theme depends on the following attributes to be set right below the title header:
+The micro-service reports the startup time by measuring the time between the `main` method entry, and the callback notification when the HTTP server has started.
 
-[source,text]
-----
-= My awesome how-to
-:page-permalink: /
-:page-github: vertx-howtos/my-howto
-----
+We can do a few runs of `java -jar build/libs/openj9-howto-all.jar` and pick the best time.
+On my machine the best I got was *639ms*.
 
-The `page-permalink` attribute ensures that `README.adoc` is rendered as `index.html` rather than `README.adoc`.
+OpenJ9 offers both an ahead-of-time compiler and a class data shared cache for improving startup time as well as reducing memory consumption.
+The first run is typical costly, but then all subsequent runs will benefit from the caches, which are also regularly updated.
 
-The `page-github` attribute must contain the joint GitHub organization and repository so it can properly link to the GitHub repository, provide a direct edition link, and more.
+The relevant OpenJ9 flags are the following:
 
-=== Jekyll configuration
+- `-Xshareclasses`: enable class sharing
+- `-Xshareclasses:name=NAME`: a name for the cache, typically one per-application
+- `-Xshareclasses:cacheDir=DIR`: a folder for storing the cache files
 
-The Jekyll configuration shall not have to be tweaked much, here is the basic configuration that we use:
+Let us have a few run of:
 
-[source,yaml]
 ----
-include::_config.yml[]
+$ java -Xshareclasses -Xshareclasses:name=sum -Xshareclasses:cacheDir=_cache -jar build/libs/openj9-howto-all.jar
 ----
 
-TIP: When running Jekyll, check the content of the `_site/` folder and adapt the `exclude` items to avoid pushing unnecessary content to the website.
+On my machine the first run takes *1098ms*, which is way more than *639ms*!
+However the next runs are all near *300ms*, with a best score of *293ms* which is very good for a JVM application start time.
 
-=== Ruby environment
+== Memory usage
 
-It is strongly recommended to use either https://github.com/rbenv/rbenv[rbenv] or https://rvm.io/[RVM] for your Ruby installation.
+=== Generate some workload
 
-You will need https://bundler.io/[Bundler] to install all required dependencies:
+We are using Locust to generate some workload.
+The `locustfile.py` file contains the code to simulate users that perform sums of random numbers:
 
+[source,python]
 ----
-$ gem install bundler
+include::locustfile.py[]
 ----
 
-You can then let _Bundler_ do the installation work:
+We can then run `locust`, and connect to http://localhost:8089 to start a test.
+Let us simulate 2000 users with a 50 new users per second hatch rate.
+This gives us about 600 requests per second.
 
-----
-$ bundle install
-----
+=== Measuring RSS
 
-From time to time you may also update dependencies:
+The https://quarkus.io/guides/performance-measure[Quarkus team has a good guide on measuring RSS].
```

Comment:
So the link to the `lengthy` article is nice :wink:

But can we get a TL;DR here to take our own measurements?